File: LMS_CMS_Backend/LMS_CMS_DAL/Models/Domains/Communication/Notification.cs
using LMS_CMS_DAL.Models.Domains.Administration;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LMS_CMS_DAL.Models.Domains.Communication
{
    public class Notification : AuditableEntity
    {
        [Key]
        public long ID { get; set; }
        public string? ImageLink { get; set; }
        public string? Text { get; set; }
        public string? Link { get; set; }
        public bool IsAllowDismiss { get; set; }

        public ICollection<NotificationSharedTo> NotificationSharedTos { get; set; } = new List<NotificationSharedTo>();
    }
}

File: LMS_CMS_Frontend/src/app/Models/Communication/notification.ts
import { NotificationSharedTo } from "./notification-shared-to";
import { UserFilters } from "./user-filters";

export class Notification {
    constructor(
        public id: number = 0,  
        public imageLink: string = '',
        public imageFile : File|null = null,
        public text: string = '',
        public link: string = '',
        public userTypeName: string = '',
        public isAllowDismiss: boolean = false,
        public userTypeID :number =0,
        public insertedByUserId :number =0,
        public notificationSharedTos : NotificationSharedTo[]=[],
        public userFilters : UserFilters= new UserFilters()
    ) {}    
} 